File: CSharp/HtmlWalker/CloneWalker.cs
using HtmlWalker.Core;
using HtmlWalker.Platform;

namespace HtmlWalker
{
    /// <summary>
    ///   This class implements a walker class to produce a clone of existing HTML.
    ///   Meant for testing the walker class and to be subclassed for uses where HTML
    ///   is copied with slight changes.
    /// </summary>
    /// <author>Robert R Flores</author>
    public class CloneWalker : DocGeneratorWalker
    {
        protected override ITag GenerateTag<T>(T adapter) => adapter.Clone(Output.Factory, currentTag);

        /// <summary>
        ///   Instanitiates the WalkerCloner that strps no tag attributes
        /// </summary>
        /// <param name="platform">the singleton that gives platform specfic implementation</param>
        public CloneWalker(WalkerPlatform.WalkerDocument document) : base(document) { }
    }
}

File: Java/htmlwalker/src/main/java/htmlwalker/CloneWalker.java
/*
 * 
 */
package htmlwalker;

import htmlwalker.core.DocGeneratorWalker;
import htmlwalker.platform.WalkerDocument;

/**
 * This class implements a walker class to produce a clone of existing HTML.
 * 
 * Meant for testing the walker class and to be subclassed for uses where HTML
 * is copied with slight changes.
 * 
 * @author Robert Flores
 */
public class CloneWalker extends DocGeneratorWalker
{
	@Override
    protected <T extends ITag> ITag generateTag(T adapter) { return adapter.clone(this.output.factory(), this.currentTag); }

	/**
	 * Instantiates a new walker cloner.
	 *
	 * @param platform - the platform implementation
	 */
	public CloneWalker(WalkerDocument document) { super(document); }
}
